api/base/testing/apicaller.go

```diff
@@ -61,31 +89,21 @@ func CheckingAPICaller(c *gc.C, args *CheckArgs, numCalls *int, err error) base.
 			if numCalls != nil {
 				*numCalls++
 			}
-			if args != nil {
-				if args.Facade != "" {
-					c.Check(facade, gc.Equals, args.Facade)
-				}
-				if args.Version != 0 {
-					c.Check(version, gc.Equals, args.Version)
-				} else if args.VersionIsZero {
-					c.Check(version, gc.Equals, 0)
-				}
-				if args.Id != "" {
-					c.Check(id, gc.Equals, args.Id)
-				} else if args.IdIsEmpty {
-					c.Check(id, gc.Equals, "")
-				}
-				if args.Method != "" {
-					c.Check(method, gc.Equals, args.Method)
-				}
-				if args.Args != nil {
-					c.Check(inArgs, jc.DeepEquals, args.Args)
-				}
-				if args.Results != nil {
-					c.Check(outResults, gc.NotNil)
-					testing.PatchValue(outResults, args.Results)
-				}
-			}
+			checkArgs(c, args, facade, version, id, method, inArgs, outResults)
+			return err
+		},
+	)
+}
+
+// PingingCheckingAPICaller returns an APICallerFunc which sends a message on the channel "called" every
+// time it recives a call, as well as check if any of the arguments passed to the APICall() method match
+// the values given in args (if args itself is not nil, otherwise no arguments are checked). The final
+// error result of the APICall() will be set to err.
+func PingingCheckingAPICaller(c *gc.C, args *CheckArgs, called chan struct{}, err error) base.APICaller {
```

Comment:
I don't like "Pinging" - it might imply it has something to do with apiserver.Ping(). How about dropping the original CheckingAPICaller and using only this one instead? It shouldn't be too difficult to change existing tests?